lighty-examples/lighty-community-restconf-netconf-app/src/main/java/io/lighty/examples/controllers/restconfapp/Main.java

```diff
@@ -100,108 +113,98 @@ public void start(String[] args, boolean registerShutdownHook) {
             LOG.info("lighty.io and RESTCONF-NETCONF started in {}", stopwatch.stop());
         } catch (IOException cause) {
             LOG.error("Main RESTCONF-NETCONF application - error reading config file: ", cause);
-        } catch (ConfigurationException | ExecutionException | InterruptedException cause) {
+        } catch (Exception cause) {
             LOG.error("Main RESTCONF-NETCONF application exception: ", cause);
+            shutdown();
         }
     }
 
     private void startLighty(ControllerConfiguration controllerConfiguration,
                              RestConfConfiguration restConfConfiguration,
                              NetconfConfiguration netconfSBPConfiguration, boolean registerShutdownHook)
-            throws ConfigurationException, ExecutionException, InterruptedException {
+            throws ConfigurationException, ExecutionException, InterruptedException, TimeoutException {
 
         //1. initialize and start Lighty controller (MD-SAL, Controller, YangTools, Akka)
         LightyControllerBuilder lightyControllerBuilder = new LightyControllerBuilder();
-        LightyController lightyController = lightyControllerBuilder.from(controllerConfiguration).build();
-        lightyController.start().get();
+        this.lightyController = lightyControllerBuilder.from(controllerConfiguration).build();
+
+        if (!this.lightyController.start().get(DEFAULT_TIMEOUT_SECONDS, TimeUnit.SECONDS)) {
+            throw new IllegalStateException("Lighty.io Controller plugin failed during startup!");
```

Comment:
Rather return true/false or use/create checked exception instead of throwing `IllegalState`

Reply:
resolved by returning true/false and then handling return value